src/guiapi/src/display_helper.cpp

```diff
@@ -197,57 +198,58 @@ void render_text_align(rect_ui16_t rc, string_view_utf8 text, const font_t *font
         size_t nLines = str2multilineUnicode(str, 0x1000U, rc_pad.w / font->w);
         // now we have '\n' in the right spots
         y += nLines * font->h;
-        render_textUnicode(rc_pad, str, font, clr0, clr1);
+        render_textUnicode(rc_pad, str, font, clr_bg, clr_fg);
 #endif
         // hack for broken text wrapper ... and for too long texts as well
         if (y < (rc_pad.y + rc_pad.h)) {
-            display::FillRect(rect_ui16(rc_pad.x, y, rc_pad.w, (rc_pad.y + rc_pad.h - y)), clr0);
-            fill_between_rectangles(&rc, &rc_pad, clr0);
+            display::FillRect(rect_ui16(rc_pad.x, y, rc_pad.w, (rc_pad.y + rc_pad.h - y)), clr_bg);
+            fill_between_rectangles(rc, rc_pad, clr_bg);
         }
-    } else {
-        // 1st pass reading the string_view_utf8 - font_meas_text also computes the number of utf8 characters (i.e. individual bitmaps) in the input string
-        uint16_t strlen_text = 0;
-        point_ui16_t wh_txt = font_meas_text(font, &text, &strlen_text);
-        if (wh_txt.x && wh_txt.y) {
-            rect_ui16_t rc_txt = rect_align_ui16(rc_pad, rect_ui16(0, 0, wh_txt.x, wh_txt.y), flags & ALIGN_MASK);
-            rc_txt = rect_intersect_ui16(rc_pad, rc_txt);
-            uint8_t unused_pxls = 0;
-            if (strlen_text * font->w > rc_txt.w) {
-                unused_pxls = rc_txt.w % font->w;
-            }
 
-            const rect_ui16_t rect_in = { rc_txt.x, rc_txt.y, uint16_t(rc_txt.w - unused_pxls), rc_txt.h };
-            fill_between_rectangles(&rc, &rect_in, clr0);
-            text.rewind();
-            // 2nd pass reading the string_view_utf8 - draw the text
-            render_text(rc_txt, text, font, clr0, clr1);
-        } else
-            display::FillRect(rc, clr0);
+        return;
+    }
+
+    // 1st pass reading the string_view_utf8 - font_meas_text also computes the number of utf8 characters (i.e. individual bitmaps) in the input string
+    uint16_t strlen_text = 0;
+    point_ui16_t wh_txt = font_meas_text(font, &text, &strlen_text);
+    if (!wh_txt.x || !wh_txt.y) {
+        display::FillRect(rc, clr_bg);
+        return;
     }
+
+    rect_ui16_t rc_txt = rect_align_ui16(rc_pad, rect_ui16(0, 0, wh_txt.x, wh_txt.y), flags & ALIGN_MASK);
+    rc_txt = rect_intersect_ui16(rc_pad, rc_txt);
+    const uint8_t unused_pxls = (strlen_text * font->w <= rc_txt.w) ? 0 : rc_txt.w % font->w;
+
+    const rect_ui16_t rect_in = { rc_txt.x, rc_txt.y, uint16_t(rc_txt.w - unused_pxls), rc_txt.h };
+    fill_between_rectangles(rc, rect_in, clr_bg);
+    text.rewind();
+    // 2nd pass reading the string_view_utf8 - draw the text
+    render_text(rc_txt, text, font, clr_bg, clr_fg);
 }
 
-void render_icon_align(rect_ui16_t rc, uint16_t id_res, color_t clr0, uint16_t flags) {
+void render_icon_align(const rect_ui16_t &rc, uint16_t id_res, color_t clr_bg, uint16_t flags) {
```

Comment:
adding a const reference here means you cannot pass an immediate value into the function. This can cause some discomfort.
AFAIK rect_uin16t is just 8B long while the pointer to rc (or a reference) is 4B long.